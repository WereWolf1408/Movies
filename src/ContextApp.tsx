import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { NetflixAppContextFunction } from './Context';
import { store } from './store/store';
import { Routers } from './Routers';
import { App } from './App';

export const ContextApp = () => {
  return (
    <Provider store={store}>
      <NetflixAppContextFunction>
        <BrowserRouter>
          <Routers />
        </BrowserRouter>
      </NetflixAppContextFunction>
    </Provider>
  );
};
