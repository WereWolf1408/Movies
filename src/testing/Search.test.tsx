import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { ContextApp } from '../ContextApp';

describe('Header Component', () => {
  it('Header snapshot test:', () => {
    const { asFragment } = render(<ContextApp />);
    expect(asFragment).toMatchSnapshot();
  });

  it(`Header 'add movie button test'`, () => {
    const { getByText } = render(<ContextApp />);
    const addMovie = getByText('+ add movie');
    expect(addMovie).toBeInTheDocument();
    userEvent.click(addMovie);
    const addMoviePopup = getByText('add movie');
    expect(addMoviePopup).toBeInTheDocument();
  });
});
