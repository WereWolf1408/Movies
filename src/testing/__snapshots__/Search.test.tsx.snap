// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header snapshot test: 1`] = `[Function]`;
